cypress/integration/jsGrowthTableDOM.spec.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+Cypress.Commands.add('clickButton',
+(selector, times) => {
+  for (let n = 0; n < times; n++) {
+    cy.get(selector).click( {force: true} );
+  }
+});
+
+describe('Table', () => {
+  beforeEach(() => {
+    cy.visit('');
+    cy.get('tbody > tr').as('rows');
+    cy.get('tbody :nth-child(1) :nth-child(n)').as('columns');
+  });
+
+  it('should have 4 rows by default', () => {
+    cy.get('@rows').should('have.length', 4);
+  });
+
+  it('should add 1 row by click', () => {
+    cy.get('.append-row').click();
+    cy.get('@rows').should('have.length', 5);
+  });
+
+  it('should have 4 columns by default', () => {
+    cy.get('@columns').should('have.length', 4);
+  });
+
+  it('should add 1 column by click', () => {
+    cy.get('.append-column').click();
+    cy.get('@columns').should('have.length', 5);
+  });
+
+  it('should have max 10 columns', () => {
+    cy.clickButton('.append-column', 7);
+    cy.get('@columns').should('have.length', 10)
+  });
+
+  it('add column button should be disabled after 10 column', () => {
+    cy.clickButton('.append-column', 7)
+    cy.get('.append-column').should('be.disabled');
+  });
+
+  it('should remove one column', () => {
+    cy.get('.remove-column').click();
+    cy.get('@columns').should('have.length', 3)
+  });
+
+  it('should have max 10 rows', () => {
+    cy.clickButton('.append-row', 7)
+    cy.get(':nth-child(7) > :nth-child(1)').should('exist');
```

Comment:
why 7, and not 10?

Reply:
Actually, it should be 6. I want to check that 10 rows are possible at all

Reply:
discussed offline: this should be 10. also need to add tbody to the selector.